lib/eslint/eslint-helpers.js

```diff
@@ -97,6 +104,105 @@ function isGlobPattern(pattern) {
     return isGlob(path.sep === "\\" ? normalizeToPosix(pattern) : pattern);
 }
 
+/**
+ * Searches a directory looking for matching glob patterns. This uses
+ * the config array's logic to determine if a directory or file should
+ * be ignored, so it is consistent with how ignoring works throughout
+ * ESLint.
+ * @param {Object} options The options for this function.
+ * @param {string} options.cwd The directory to search.
+ * @param {Array<string>} options.patterns An array of glob patterns
+ *      to match.
+ * @param {FlatConfigArray} options.configs The config array to use for
+ *      determining what to ignore.
+ * @returns {Promise<Array<string>>} An array of matching file paths
+ *      or an empty array if there are no matches.
+ */
+async function globSearch({ cwd, patterns, configs }) {
+
+    if (patterns.length === 0) {
+        return [];
+    }
+
+    const matchers = patterns.map(pattern => new minimatch.Minimatch(pattern));
+
+    return (await doFsWalk(cwd, {
+
+        deepFilter(entry) {
+            return !configs.isDirectoryIgnored(entry.path);
+        },
+
+        entryFilter(entry) {
```

Comment:
`entryFilter` determines which entries will show up in the results, so I filter out directories and check patterns here.